### DISCUSSION TITLE
Design questions

### DISCUSSION THREAD

**Discussion by drewstone** [open-web3-stack/open-runtime-module-library#338] [Q&A] [answered]
Hey all,

Looking to use this repo as it has nice nice token library that supports multiple assets. I have some questions that are probably easy to answer, and some likely obvious, but thought I'd ask.

- Is it correct that the CurrencyAdapter only works for a single currency? How should one go about burning an arbitrary currency ID?

- Is there any way to mint assets using `orml-tokens`? We're interested in using this potentially in conjunction with ChainBridge and would need a way to mint assets, though perhaps the substrate assets is better suited for this.

**Comment by xlc** [comment, marked as answer]
Use `orml_traits::MultiCurrency` if you want to deal with multiple currency. It also supports deposit / withdraw which is basically mint / burn. They are not exposed as extrinsic but you can call them in your pallet.
